MTA-6889 CLI refactored commands (#372)
migtools/mta-documentation: docs/topics/mta-cli/con_cli-command-structure.adoc

// Module included in the following assemblies:
//
// * assemblies/cli-guide/assembly_about-the-command-line-tool.adoc

:_mod-docs-content-type: CONCEPT
[id="cli-command-structure_{context}"]
= CLI command structure

[role="_abstract"]

> To simplify your understanding of capabilities in the {ProductFullName}  command-line interface (CLI) and to better classify tasks you can do, the subcommands and options are reclassified under the main commands.

Reviewer: ```suggestion
To help you easily classify tasks, the {ProductFullName} command-line interface (CLI) organizes subcommands and options under main commands.
```

There are two main issues here: wordiness and passive voice.

* The introductory phrase "To simplify your understanding of capabilities... and to better classify tasks you can do" contains a lot of fluff. CQA 2.1 strictly mandates the elimination of fluff by avoiding long introductions and shortening unnecessarily long sentences.

* The phrase "the subcommands and options are reclassified" is in the passive voice. IBM Style requires using the active voice as much as possible because it is clearer, shorter, and more direct.

Flagging as this will get caught in the next CQA report run

> A command performs a specific task to prepare applications for migration. You can explore two new commands for providers and rules related tasks.

Reviewer: ```suggestion
For example, you can use commands to manage providers and rules.
```

IBM Style strictly prohibits using the word "new" to describe features, functions, or commands (unless in a specific release note or "What's new" topic) because the documentation might remain in circulation long after the feature is established or obsolete. Additionally, "rules related tasks" is slightly awkward.

Author: Hey Andy, I have removed this statement since this sentence is followed by a list of commands with descriptions. How do you feel about it?

You can use the following commands in the {ProductShortName} CLI:

`mta-cli analyze`::

> Contains options to specify runtime settings for the analysis type or mode, single or multiple input applications, dependencies, rules, source or target technologies, proxy URL, a local or a containerized run, and the analysis output.

Reviewer: ```suggestion
Provides options to specify runtime settings for the analysis type or mode, one or more input applications, dependencies, rules, source or target technologies, proxy URL, a local or a containerized run, and the analysis output.
```

`mta-cli rules`::

> Contains subcommands to list source and target technologies and test rules available for analyzing the source and target technologies. You can test default and custom rulesets.

Reviewer: ```suggestion
Provides subcommands to list source and target technologies and to test the rules that are available to analyze them. You can test default and custom rulesets.
```

`mta-cli provider`::

> Contains subcommands to list the language-specific providers. Providers work with language server protocol (LSP) servers to parse the application source code to find specific problem patterns that you defined in the rules.

Reviewer: ```suggestion
Provides subcommands to list the language-specific providers. Providers work with language server protocol (LSP) servers to parse the application source code and find specific problem patterns that you define in the rules.
```

tense consistency

`mta-cli openrewrite`::

> Contains subcommands with options to use OpenRewrite recipes to transform your codebase by updating Java libraries and framework for a target technology.

Reviewer: ```suggestion
Provides subcommands and options to use OpenRewrite recipes to transform your codebase by updating Java libraries and frameworks for a target technology.
```

`mta-cli config`::

> Contains subcommands to log in to the {ProductShortName} Hub by using a secure or an insecure connection to sync, download, and manage a reusable analysis configuration file called profile in your local system.

Reviewer: ```suggestion
Provides subcommands to log in to the {ProductShortName} Hub by using a secure or insecure connection to sync, download, and manage a reusable analysis configuration file, called a profile, on your local system.
```

`mta-cli discover`::

> Contains a subcommand with options to generate a local YAML manifest from application manifests in the Cloud Foundry (CF) source platform. You can filter CF applications by organization, space, and a specific application name. You can also redact sensitive information such as username and secret in the discovery manifest.

Reviewer: ```suggestion
Provides a subcommand and options to generate a local YAML manifest from application manifests in the Cloud Foundry (CF) source platform. You can filter CF applications by organization, space, and a specific application name. You can also redact sensitive information, such as usernames and secrets, in the discovery manifest.
```

`mta-cli generate`::

> Contains a subcommand with options to generate Kubernetes and non-Kubernetes deployment manifests by using Helm templates and value overrides.

Reviewer: ```suggestion
Provides a subcommand and options to generate Kubernetes and non-Kubernetes deployment manifests by using Helm templates and value overrides.
```

[IMPORTANT]
====
The `config`, `discover`, and `generate` commands feature no modifications to their associated flags and options.

> ====

Reviewer: ```suggestion
[NOTE]
====
The options for the `config`, `discover`, and `generate` commands are unchanged.
====
```

Changed [IMPORTANT] to [NOTE]. The Red Hat SSG strictly reserves IMPORTANT for "advisory information essential to the completion of a task". A historical statement about unchanged commands is supplemental information, which better fits a NOTE
